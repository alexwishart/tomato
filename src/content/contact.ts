export const contactContent = {
  title: 'Keep up to date on the tomato',
  subtitle:
    'Enter your name and email address below to receive weekly updates on the status of the tomato',
  inputFields: [
    { value: 'name', label: 'Name' },
    { value: 'email', label: 'E-mail' },
  ],
  submitButtonLabel: 'Submit',
};
