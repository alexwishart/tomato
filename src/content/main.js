import tomatoImage from '../images/tomato.jpg';

export const mainContent = {
  title: 'The Tomato',
  image: tomatoImage,
};
