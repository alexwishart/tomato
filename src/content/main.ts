import tomatoImage from '../images/tomato.jpg';

export const mainContent = {
  title: 'The Tomato',
  image: { src: tomatoImage, alt: 'Image of tomato' },
  content: {
    title: 'Varieties',
    paragraphs: [
      "There are around 7500 tomato varieties grown for various purposes. Heirloom tomatoes are becoming increasingly popular, particularly among home gardeners and organic producers, since they tend to produce more interesting and flavorful crops at the cost of disease resistance and productivity. In 1973, Israeli scientists developed the world's first long shelf-life commercial tomato varieties.",
      'Tomato varieties are roughly divided into several categories, based mostly on shape and size.',
    ],
    varieties: [
      '"Slicing"/"globe" tomatoes',
      'Beefsteak tomatoes',
      'Oxheart tomatoes',
      'Plum tomatoes',
      'Pear tomatoes',
      'Cherry tomatoes',
      'Grape tomatoes',
      'Campari tomatoes',
    ],
  },
};
