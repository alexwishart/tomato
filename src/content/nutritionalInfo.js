export const nutritionalInfoContent = {
  title: 'Red tomatoes, raw',
  subtitle: 'Nutritional value per 100 g (3.5 oz)',
  table: {
    header: ['Energy', '74kJ (18 kcal)'],
    rows: [
      ['Carbohydrates', '3.9 g'],
      ['- Sugars', '2.6 g'],
      ['- Dietary fiber', '1.2 g'],
      ['Fat', '0.2 g'],
      ['Protein', '0.9 g'],
      ['Water', '94.5 g'],
      ['Vitamin A equiv.', '42 µg (5%)'],
      ['- lutein and zeaxanthin', '123 µg'],
      ['Vitamin C', '14 mg (17%)'],
      ['Vitamin E', '0.54 mg (4%)'],
      ['Potassium', '237 mg (5%)'],
    ],
  },
  footer: {
    paragraphs: [
      'Percentages are relative to US recommendations for adults',
      'Source:',
    ],
    link: {
      text: 'USDA Nutrient Database',
      href: 'http://www.nal.usda.gov/fnic/foodcomp/search/',
    },
  },
};
