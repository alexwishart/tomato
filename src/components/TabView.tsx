import React, { useState } from 'react';
import styled from 'styled-components';
import { TabInfo } from '../content/tabInfo';

const TabView: React.FC<{ tabs: TabInfo[] }> = (props: { tabs: TabInfo[] }) => {
  const [activeTab, setActiveTab] = useState(0);

  return (
    <>
      <Tabs role="tablist">
        {props.tabs.map((tab, i) => {
          return (
            <Tab role="presentation" key={tab.title}>
              <TabButton
                role="tab"
                onClick={() => {
                  setActiveTab(i);
                }}
                active={i === activeTab}
              >
                {tab.title}
              </TabButton>
            </Tab>
          );
        })}
      </Tabs>
      <TabContent role="tabpanel">{props.tabs[activeTab].content}</TabContent>
    </>
  );
};

const Tabs = styled.ul`
  width: 100%;
  display: flex;
  justify-content: space-evenly;
  padding: 0;
`;

const Tab = styled.li`
  width: 100%;
  height: 2rem;
  padding: 0;
  margin: 0;
  list-style-type: none;
`;

const TabButton = styled.button<{ active: boolean }>`
  height: 100%;
  width: 100%;
  padding: 0;
  font: inherit;
  cursor: pointer;
  border: none;
  outline: 0.1rem solid black;

  background: ${(props) => (props.active ? 'none' : 'light gray')};
`;

const TabContent = styled.div`
  height: 10rem;
  padding: 0.5rem;
  outline: 0.1rem solid black;

  overflow-y: scroll;
`;

export default TabView;
