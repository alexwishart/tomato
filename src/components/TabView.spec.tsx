import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import TabView from './TabView';
import { act } from 'react-dom/test-utils';
import { TabInfo } from '../content/tabInfo';

describe('The TabView component', () => {
  it('should render correctly (with the first tab active)', () => {
    expect(render(<TabView tabs={testTabs} />)).toMatchSnapshot();
  });

  it('should render correctly when a different tab is clicked', () => {
    const { getAllByRole, queryByText } = render(<TabView tabs={testTabs} />);

    const firstTab = getAllByRole('tab')[0];
    const secondTab = getAllByRole('tab')[1];

    act(() => {
      fireEvent.click(secondTab);
    });

    expect(firstTab).toHaveStyleRule('background', 'light gray');
    expect(secondTab).toHaveStyleRule('background', 'none');
    expect(queryByText(testTabs[0].content)).toBeFalsy();
    expect(queryByText(testTabs[1].content)).toBeTruthy();
  });
});

const testTabs: TabInfo[] = [
  { title: '1', content: 'content 1' },
  { title: '2', content: 'content 2' },
  { title: '3', content: 'content 3' },
];
