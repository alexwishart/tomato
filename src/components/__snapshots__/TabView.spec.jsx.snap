// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TabView component should render correctly (with the first tab active) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  padding: 0;
}

.c1 {
  width: 100%;
  height: 2rem;
  padding: 0;
  margin: 0;
  list-style-type: none;
}

.c2 {
  height: 100%;
  width: 100%;
  padding: 0;
  font: inherit;
  cursor: pointer;
  border: none;
  outline: 0.1rem solid black;
  background: none;
}

.c3 {
  height: 100%;
  width: 100%;
  padding: 0;
  font: inherit;
  cursor: pointer;
  border: none;
  outline: 0.1rem solid black;
  background: light gray;
}

<body>
    <div>
      <ul
        class="c0"
        role="tablist"
      >
        <li
          class="c1"
          role="presentation"
        >
          <button
            class="c2"
            role="tab"
          >
            Nutrition
          </button>
        </li>
        <li
          class="c1"
          role="presentation"
        >
          <button
            class="c3"
            role="tab"
          >
            Health benefits
          </button>
        </li>
        <li
          class="c1"
          role="presentation"
        >
          <button
            class="c3"
            role="tab"
          >
            Storage
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-bdfBwQ dcSSuq"
      role="tablist"
    >
      <li
        class="sc-gsTCUz bVqLfQ"
        role="presentation"
      >
        <button
          class="sc-dlfnbm bxWKmq"
          role="tab"
        >
          Nutrition
        </button>
      </li>
      <li
        class="sc-gsTCUz bVqLfQ"
        role="presentation"
      >
        <button
          class="sc-dlfnbm kdjINT"
          role="tab"
        >
          Health benefits
        </button>
      </li>
      <li
        class="sc-gsTCUz bVqLfQ"
        role="presentation"
      >
        <button
          class="sc-dlfnbm kdjINT"
          role="tab"
        >
          Storage
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
